networking_nsxv3/plugins/ml2/drivers/nsxv3/agent/provider_nsx_policy.py

```diff
@@ -448,11 +447,24 @@ def __init__(self, payload: Payload = Payload(), zone_id: str = ""):
             self._ensure_default_l3_policy()
         self._setup_default_app_drop_logged_section()
 
-    def _load_zone(self):
+    def _get_tz(self) -> dict or None:
+        for tz in self.client.get_all(path=API.SEARCH_QUERY, params={"query": API.SEARCH_Q_TRANSPORT_ZONES.format(self.zone_name)}):
+            if tz.get("display_name") == self.zone_name:
+                return tz
+        return None
+
+    def _load_zones(self):
         LOG.info("Looking for TransportZone with name %s.", self.zone_name)
-        for zone in self.client.get_all(path=API.SEARCH_QUERY, params={"query": API.SEARCH_Q_TRANSPORT_ZONES.format(self.zone_name)}):
-            if zone.get("display_name") == self.zone_name:
-                return zone.get("id")
+
```

Comment:
return self.tz()

Reply:
this is concrete implementation of an abstract method which expects tuple as return